packages/teams-js/src/public/dialog.ts

```diff
@@ -111,183 +88,383 @@ export namespace dialog {
+   *
+   * @internal
+   */
+  export function getDialogInfoFromBotUrlDialogInfo(botUrlDialogInfo: BotUrlDialogInfo): DialogInfo {
+    const dialogInfo: DialogInfo = {
+      url: botUrlDialogInfo.url,
+      height: botUrlDialogInfo.size ? botUrlDialogInfo.size.height : DialogDimension.Small,
+      width: botUrlDialogInfo.size ? botUrlDialogInfo.size.width : DialogDimension.Small,
+      title: botUrlDialogInfo.title,
+      fallbackUrl: botUrlDialogInfo.fallbackUrl,
+      completionBotId: botUrlDialogInfo.completionBotId,
+    };
+    return dialogInfo;
   }
 
   /**
-   * Checks if dialog module is supported by the host
+   * @hidden
+   * Hide from docs
+   * --------
+   * Convert AdaptiveCardDialogInfo to DialogInfo to send the information to host in {@linkcode adaptiveCard.open} API.
    *
-   * @returns boolean to represent whether dialog module is supported
+   * @internal
    */
-  export function isSupported(): boolean {
-    return runtime.supports.dialog ? true : false;
+  export function getDialogInfoFromBotAdaptiveCardDialogInfo(
+    botAdaptiveCardDialogInfo: BotAdaptiveCardDialogInfo,
+  ): DialogInfo {
+    const dialogInfo: DialogInfo = {
+      card: botAdaptiveCardDialogInfo.card,
+      height: botAdaptiveCardDialogInfo.size ? botAdaptiveCardDialogInfo.size.height : DialogDimension.Small,
+      width: botAdaptiveCardDialogInfo.size ? botAdaptiveCardDialogInfo.size.width : DialogDimension.Small,
+      title: botAdaptiveCardDialogInfo.title,
+      completionBotId: botAdaptiveCardDialogInfo.completionBotId,
+    };
+    return dialogInfo;
   }
 
   /**
-   * Namespace to update the dialog
+   * Subcapability for interacting with adaptive card dialogs
    */
-  export namespace update {
+  export namespace adaptiveCard {
     /**
-     * Update dimensions - height/width of a dialog.
+     * Allows app to open an adaptive card based dialog.
      *
-     * @param dimensions - An object containing width and height properties.
+     * @remarks
+     * This function cannot be called from inside of a dialog
+     *
+     * @param adaptiveCardDialogInfo - An object containing the parameters of the dialog module.
+     * @param submitHandler - Handler that triggers when a dialog calls the {@linkcode submit} function or when the user closes the dialog.
      */
-    export function resize(dimensions: DialogSize): void {
-      ensureInitialized(FrameContexts.content, FrameContexts.sidePanel, FrameContexts.task, FrameContexts.meetingStage);
+    export function open(adaptiveCardDialogInfo: AdaptiveCardDialogInfo, submitHandler?: DialogSubmitHandler): void {
+      ensureInitialized(FrameContexts.content, FrameContexts.sidePanel, FrameContexts.meetingStage);
       if (!isSupported()) {
         throw errorNotSupportedOnPlatform;
       }
-      sendMessageToParent('tasks.updateTask', [dimensions]);
+
+      getVersion().then(hostVersion => {
```

Comment:
Does this mean a `NOT_SUPPORTED_ON_PLATFORM` kind of error can be thrown two different ways? If `isSupported` returns true, it'll be a simple throw (a synchronous one), but if it's the version that's not good enough, the function won't throw per se, and instead the promise we return will be rejected? Seems a bit messy?

Reply:
Agreed, I will clean this up and make them both synchronous